tests/sonata.cpp

```diff
@@ -0,0 +1,181 @@
+    BOOST_CHECK_EQUAL(node_ids[0], 0);
+    BOOST_CHECK_EQUAL(node_ids[10], 0);
+    BOOST_CHECK_EQUAL(node_ids[19], 1);
+}
+
+BOOST_AUTO_TEST_CASE(sonata_getNodeGroupIndices)
+{
+    brion::Nodes nodes(TEST_MORPHOLOGY_URI);
+    const auto node_ids = nodes.getNodeGroupIndices(POPULATION_NAME);
+    BOOST_CHECK_EQUAL(node_ids.size(), 20);
+    BOOST_CHECK_EQUAL(node_ids[0], 9);
+    BOOST_CHECK_EQUAL(node_ids[10], 9);
+    BOOST_CHECK_EQUAL(node_ids[19], 4);
+}
+
+BOOST_AUTO_TEST_CASE(sonata_getNodeTypes)
+{
+    brion::Nodes nodes(TEST_MORPHOLOGY_URI);
+    const auto node_ids = nodes.getNodeTypes(POPULATION_NAME);
+    BOOST_CHECK_EQUAL(node_ids.size(), 20);
+    BOOST_CHECK_EQUAL(node_ids[0], 0);
+    BOOST_CHECK_EQUAL(node_ids[10], 2);
+    BOOST_CHECK_EQUAL(node_ids[19], 1);
+}
+
+/////////////////////////////////////////////////////////////////////////////
+
+BOOST_AUTO_TEST_CASE(sonata_nodeGroup_getAttributeNames)
+{
+    brion::Nodes nodes(TEST_MORPHOLOGY_URI);
+    auto group = nodes.openGroup(POPULATION_NAME, 0);
+    const auto names = group.getAttributeNames();
+    BOOST_CHECK_EQUAL(names.size(), 8);
+    BOOST_CHECK_EQUAL(names[1], "morphology_file");
+    BOOST_CHECK_EQUAL(names[2], "rotation_angle_x");
+    BOOST_CHECK_EQUAL(names[3], "rotation_angle_y");
+}
+
+BOOST_AUTO_TEST_CASE(sonata_nodeGroup_getDynamicParameterNames)
+{
+    brion::Nodes nodes(TEST_MORPHOLOGY_URI);
+    auto group = nodes.openGroup(POPULATION_NAME, 0);
+    const auto names = group.getDynamicParameterNames();
+    BOOST_CHECK_EQUAL(names.size(), 1);
+    BOOST_CHECK_EQUAL(names[0], "my_dataset");
+}
+
+BOOST_AUTO_TEST_CASE(sonata_nodeGroup_getDynamicParameter)
+{
+    brion::Nodes nodes(TEST_MORPHOLOGY_URI);
+    auto group = nodes.openGroup(POPULATION_NAME, 0);
+
+    const auto my_dataset =
+        group.getDynamicParameter<brion::floats>("my_dataset");
+    BOOST_CHECK_EQUAL(my_dataset.size(), 10);
+    BOOST_CHECK(fuzz_eq(my_dataset[0], 0.1));
+}
+
+BOOST_AUTO_TEST_CASE(sonata_nodeGroup_getNumberOfNodes)
+{
+    brion::Nodes nodes(TEST_MORPHOLOGY_URI);
+    auto group = nodes.openGroup(POPULATION_NAME, 0);
+    const auto num_nodes = group.getNumberOfNodes();
+    BOOST_CHECK_EQUAL(num_nodes, 8);
+}
+
+BOOST_AUTO_TEST_CASE(sonata_nodeGroup_getAttribute)
+{
+    brion::Nodes nodes(TEST_MORPHOLOGY_URI);
+    auto group = nodes.openGroup(POPULATION_NAME, 0);
+
+    const auto rotation_angle_x =
+        group.getAttribute<brion::floats>("rotation_angle_x");
+    const auto rotation_angle_y =
+        group.getAttribute<brion::floats>("rotation_angle_y");
+    const auto rotation_angle_z =
+        group.getAttribute<brion::floats>("rotation_angle_z");
+    const auto x = group.getAttribute<brion::floats>("x", 2, 12);
+    const auto y = group.getAttribute<brion::floats>("y", 3, 12);
+    const auto z = group.getAttribute<brion::floats>("z", 6, 10);
+
+    BOOST_CHECK_EQUAL(rotation_angle_x.size(), 10);
+    BOOST_CHECK(fuzz_eq(rotation_angle_x[0], 0.1));
```

Comment:
Use BOOST_CHECK_EQUAL and use 0.1f. There are not arithmetic operations or conversions involved, so the values should be bitwise equal.
And by the way, there's a BOOST_CHECK_CLOSE.